src/tests/unit/tests/background/stores/assessment-card-selection-store.test.ts

```diff
@@ -85,6 +87,181 @@ describe('AssessmentCardSelectionStore Test', () => {
+                initialState,
+                null,
+                null,
+                true,
+                '',
+            );
+            testObject.initialize();
+
+            expect(testObject.getState()).toEqual(expectedState);
+        });
+
+        it('sets the state to default state if no param or persisted state', () => {
+            const testObject = new AssessmentCardSelectionStore(
+                new AssessmentCardSelectionActions(),
+                null,
+                null,
+                null,
+                false,
+                '',
+            );
+            testObject.initialize();
+
+            expect(testObject.getState()).toEqual({});
+        });
+    });
+
+    describe('onAssessmentStoreChanged', () => {
+        it('sets the store state with assessment data', async () => {
+            const payload: AssessmentStoreChangedPayload = {
+                assessmentStoreData: createAssessmentStoreDataWithStatus(ManualTestStatus.FAIL),
+            };
+
+            const storeTester =
+                createStoreForAssessmentCardSelectionActions(
+                    'assessmentStoreChanged',
+                ).withActionParam(payload);
+            await storeTester.testListenerToBeCalledOnce(null, expectedState);
+        });
+
+        it('does not create cards when assessment data contains no failure instances', async () => {
+            const payload: AssessmentStoreChangedPayload = {
+                assessmentStoreData: createAssessmentStoreDataWithStatus(ManualTestStatus.PASS),
+            };
+
+            expectedState['testKey1'].rules = null;
+            expectedState['testKey2'].rules = null;
+
+            const storeTester =
+                createStoreForAssessmentCardSelectionActions(
+                    'assessmentStoreChanged',
+                ).withActionParam(payload);
+            await storeTester.testListenerToBeCalledOnce(null, expectedState);
+        });
+
+        it('sets the store state to {} when assessment store data is empty', async () => {
+            const payload: AssessmentStoreChangedPayload = {
+                assessmentStoreData: {} as AssessmentStoreData,
+            };
+
+            expectedState = {};
+
+            const storeTester =
+                createStoreForAssessmentCardSelectionActions(
+                    'assessmentStoreChanged',
+                ).withActionParam(payload);
+            await storeTester.testListenerToBeCalledOnce(null, expectedState);
+        });
+
+        it('does nothing when there is no payload', async () => {
+            const storeTester =
+                createStoreForAssessmentCardSelectionActions(
+                    'assessmentStoreChanged',
+                ).withActionParam(null);
+            await storeTester.testListenerToNeverBeCalled(initialState, expectedState);
+        });
+
+        it('does nothing when there is no assessment store data in payload', async () => {
```

Comment:
nit: Could probably do a it.each here with all of the cases where we expect the initial and expected state to end up the same- including when the assessmentStoreData.assessments are null and when they are empty.